flake/dev/nixpkgs-and-dev-nixpkgs-version-consistency.nix

```diff
@@ -0,0 +1,15 @@
+{ inputs, ... }:
+{
+  perSystem =
+    { lib, pkgs, ... }:
+    {
+      checks.nixpkgs-and-dev-nixpkgs-version-consistency =
+        let
+          dev-nixpkgs = inputs.dev-nixpkgs.narHash;
+          nixpkgs = inputs.nixpkgs.narHash;
+        in
+        lib.throwIf (nixpkgs != dev-nixpkgs)
+          "inconsistent nixpkgs (${nixpkgs}) and dev-nixpkgs (${dev-nixpkgs}) versions"
+          pkgs.emptyDirectory;
```

Comment:
This assertion will trigger when evaluating the test, not when running the test.

See https://github.com/nix-community/stylix/pull/1289#issuecomment-3036946420

Not necessarily a blocker, but generally if you have a test derivation you should do your assertions at build time, not eval time.

E.g. `nix flake show` may evaluate this to print the derivation name, which would inadvertently also run the assertion.